sdk/src/models/record-scanner/ownedFilter.ts

```diff
@@ -0,0 +1,24 @@
+import { RecordSearchParams } from "../record-provider/recordSearchParams";
+import { RecordsFilter } from "./recordsFilter";
+import { RecordsResponseFilter } from "../record-provider/recordsResponseFilter";
+
+/**
+ * OwnedFilter is an extension of RecordSearchParams that represents a filter for scanning owned records.
+ * 
+ * @example
+ * const ownedFilter: OwnedFilter = {
+ *     unspent: true,
+ *     nonces: ["3077450429259593211617823051143573281856129402760267155982965992208217472983group"],
+ *     decrypt: true,
+ *     filter: {
+ *         program: "credits.aleo",
+ *         record: "credits",
+ *     },
+ * }
+ */ 
+export interface OwnedFilter extends RecordSearchParams {
+    decrypt?: boolean;
+    filter?: RecordsFilter;
+    responseFilter?: RecordsResponseFilter;
```

Comment:
```suggestion
    responseFilter?: RecordsResponseFilter;
```

We need to create a new type for the `OwnedResponseFilter` to match this Rust type and change it in this interface:

```rust
pub struct OwnedResponseFilter {
    pub commitment: Option<bool>,
    pub owner: Option<bool>,
    pub tag: Option<bool>,
    pub sender: Option<bool>,
    pub spent: Option<bool>,
    pub record_ciphertext: Option<bool>,
    pub block_height: Option<bool>,
    pub block_timestamp: Option<bool>,
    pub output_index: Option<bool>,
    pub record_name: Option<bool>,
    pub function_name: Option<bool>,
    pub program_name: Option<bool>,
    pub transition_id: Option<bool>,
    pub transaction_id: Option<bool>,
    pub transaction_index: Option<bool>,
    pub transition_index: Option<bool>,
}
```